buildSrc/src/main/java/com/r3/csde/CordaStatusQueries.java

```diff
@@ -1,64 +1,117 @@
 package com.r3.csde;
 
+import kong.unirest.HttpResponse;
 import kong.unirest.JsonNode;
 import kong.unirest.Unirest;
 import kong.unirest.json.JSONArray;
 import kong.unirest.json.JSONObject;
-import kong.unirest.HttpResponse;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
 
 // todo: This class needs refactoring, see https://r3-cev.atlassian.net/browse/CORE-11624
 public class CordaStatusQueries {
 
     ProjectContext pc;
-    public CordaStatusQueries(ProjectContext _pc){ pc = _pc; }
 
+    public CordaStatusQueries(ProjectContext _pc) {
+        pc = _pc;
+    }
+
+    public HttpResponse<JsonNode> getCpiInfo() {
+        Unirest.config().verifySsl(false);
+        return Unirest.get(pc.baseURL + "/api/v1/cpi/")
+                .basicAuth(pc.rpcUser, pc.rpcPasswd)
+                .asJson();
+    }
 
     public HttpResponse<JsonNode> getVNodeInfo() {
         Unirest.config().verifySsl(false);
         return Unirest.get(pc.baseURL + "/api/v1/virtualnode/")
                 .basicAuth(pc.rpcUser, pc.rpcPasswd)
                 .asJson();
     }
+
+    // cpiName, cpiVersion
+    public void listCPIs() {
+        HttpResponse<JsonNode> cpiResponse = getCpiInfo();
+
+        JSONArray cpisJson = (JSONArray) cpiResponse.getBody().getObject().get("cpis");
+
+        List<List<String>> lines = new LinkedList<>();
```

Comment:
I think `List<List<String>>` is an overkill ... 
Isn't there some List<Tuple<String>> ? 
Or pair (like c++) ?

Reply:
The outer List is the list of lines in the table.
The inner List is the list of columns/cells in one line of the table.

If I make it a Pair this means that we're not allowed to have more than 2 columns in the table. If you check method "listVNodes()", specifically line 78, there is an example of table with 3 columns
If I use Tuple 
- either this means to put a restriction that all tables have the same number of columns. At the moment "listCPIs" is with 2 and "listVNodes" is with 3;
- or I'm getting "Unchecked assignment" warnings

So even "List<List<String>>" looks complex, it's the best one